core/src/test/scala/unit/kafka/server/ReplicationQuotasTest.scala

```diff
@@ -156,31 +151,24 @@ class ReplicationQuotasTest extends ZooKeeperTestHarness {
 
     val throttledTook = System.currentTimeMillis() - start
 
-    //Check the recorded throttled rate is what we expect
-    if (leaderThrottle) {
-      (100 to 105).map(brokerFor(_)).foreach { broker =>
-        val metricName = broker.metrics.metricName("byte-rate", LeaderReplication.toString, "Tracking byte-rate for" + LeaderReplication)
-        val measuredRate = broker.metrics.metrics.asScala(metricName).value()
-        info(s"Broker:${broker.config.brokerId} Expected:$throttle, Recorded Rate was:$measuredRate")
-        assertEquals(throttle, measuredRate, percentError(25, throttle))
-      }
-    } else {
-      (106 to 107).map(brokerFor(_)).foreach { broker =>
-        val metricName = broker.metrics.metricName("byte-rate", FollowerReplication.toString, "Tracking byte-rate for" + FollowerReplication)
-        val measuredRate = broker.metrics.metrics.asScala(metricName).value()
-        info(s"Broker:${broker.config.brokerId} Expected:$throttle, Recorded Rate was:$measuredRate")
-        assertEquals(throttle, measuredRate, percentError(25, throttle))
-      }
-    }
-
     //Check the times for throttled/unthrottled are each side of what we expect
-    info(s"Unthrottled took: $unthrottledTook, Throttled took: $throttledTook, for expeted $expectedDuration secs")
-    assertTrue(s"Unthrottled replication of ${unthrottledTook}ms should be < ${expectedDuration * 1000}ms",
-      unthrottledTook < expectedDuration * 1000)
-    assertTrue((s"Throttled replication of ${throttledTook}ms should be > ${expectedDuration * 1000}ms"),
-      throttledTook > expectedDuration * 1000)
-    assertTrue((s"Throttled replication of ${throttledTook}ms should be < ${expectedDuration * 1500}ms"),
-      throttledTook < expectedDuration * 1000 * 1.5)
+    val throttledLowerBound = expectedDuration * 1000 * 0.9
+    val throttledUpperBound = expectedDuration * 1000 * 3
+    assertTrue(s"Expected $unthrottledTook < $throttledLowerBound", unthrottledTook < throttledLowerBound)
+    assertTrue(s"Expected $throttledTook > $throttledLowerBound", throttledTook > throttledLowerBound)
+    assertTrue(s"Expected $throttledTook < $throttledUpperBound", throttledTook < throttledUpperBound)
+
+    // Check the rate metric matches what we expect.
+    // In a short test the brokers can be read unfairly, so assert against the average
+    val rateUpperBound = throttle * 1.1
+    val rateLowerBound = throttle * 0.5
+    val rate = if (leaderThrottle) avRate(LeaderReplication, 100 to 105) else avRate(FollowerReplication, 106 to 107)
+    assertTrue(s"Expected ${rate} < $rateUpperBound", rate < rateUpperBound)
+    assertTrue(s"Expected ${rate} > $rateLowerBound", rate > rateLowerBound)
```

Comment:
Not sure I understand the second qu. If you went in and altered the throttle, whilst the test was running, it would fail, but I'm not sure that's what you mean.